app/models/host/discovered.rb

```diff
@@ -247,4 +247,20 @@ def create_notification
   def delete_notification
     ForemanDiscovery::UINotifications::DestroyHost.deliver!(self)
   end
+
+  def notification_recipients_ids
+    organization_users = if organization.ignore_types.include? 'User'
+                           User.unscoped.all
+                         else
+                           organization.users
+                         end
+
+    org_recipients = organization_users.find_all do |user|
+      user.allowed_to? "create_hosts"
+    end
```

Comment:
the whole login until this line would be probably worth extracting into new method like `Organization#users_with_permission(name)`. While it will also find users with filtered permissions and might result in notifying a user who at the end can't create the host, it's still good guess. I know it would be useful e.g. in Katello to display notification for users who can refresh manifests.